### PR TITLE
[core] Status JSON and Flare options

### PR DESCRIPTION
**Pull request by JohnLZeller**
This PR has the agent drop json for use by the new gui. In addition, I've also added some command line options to the flare command, to make it usable without interaction at the command line.

### CODE REVIEW

File: checks/check_status.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 from collections import defaultdict
 import cPickle as pickle
 import datetime
+import json
```
Comment (remh) [inline checks/check_status.py]: simplejson

**Comment by remh** [comment]
Can you describe a bit more how and why those features will be used ?

**Comment by JohnLZeller** [comment]
Sure! So, I've made two changes here.

Firstly, currently we pickle python objects for that status of the collector, forwarder, and dogstatsd. The current GUI loads these pickled objects, and calls `.to_dict()` on each of them, before accessing its' attributes. With the changes here, the pickling still happens (can be removed later), but we are also simply dumping the dictionaries for each of these status objects, as json. There are now JSON dumps for the collector, forwarder, dogstatsd, and an additional one with basic agent stats like version, architecture, etc. That last one doesn't need to be dumped as often, but this gets the information accessible for now.

Secondly, the flare command currently only takes a single command line arg, for Zendesk ticket number. I've added option parameters for supplying an email, and for skipping any interactive command line stuff (such as confirming that you'd really, really like to send a flare). This makes it very easy to call the flare command from the node application (the gui).

**Comment by JohnLZeller** [comment]
@hkaj would you mind taking a look?

**Comment by gmmeyer** [comment]
😢

**Pull request closed without merging**